datadog_checks_base/datadog_checks/base/utils/db/query.py

```diff
@@ -52,8 +52,8 @@ def compile(
         query = self.query_data.get('query')
         if not query:
             raise ValueError('field `query` for {} is required'.format(query_name))
-        elif not isinstance(query, str):
-            raise ValueError('field `query` for {} must be a string'.format(query_name))
+        elif not isinstance(query, str) and not isinstance(query, dict):
+            raise ValueError('field `query` for {} must be a string or a mapping'.format(query_name))
```

Comment:
Let's do this in a separate PR https://datadoghq.dev/integrations-core/guidelines/pr/#separation-of-concerns